In [1]:
import pandas as pd
import seaborn as sns

import matplotlib.pyplot as plt
# import missingno as msno

import warnings

%matplotlib inline

warnings.filterwarnings("ignore")
plt.style.use("ggplot")
plt.style.use('dark_background')


In [2]:
file = 'netflix_titles.csv'
movies = pd.read_csv(file)
movies.shape

(6234, 12)

In [3]:
for i in movies.columns:
    null_rate = movies[i].isna().sum() / len(movies) * 100 
    if null_rate > 0 :
        print(f"{i}'s null rate : {null_rate} %")

director's null rate : 31.584857234520374 %
cast's null rate : 9.143407122232915 %
country's null rate : 7.635547000320821 %
date_added's null rate : 0.17645171639396856 %
rating's null rate : 0.16041065126724416 %


There's a total of 6234 videos available and its features are director, cast & country.

As we can see from above directory, cast & country has lots of missing data.

In [4]:
movies.head(5)

show_id     type                                    title  \
0  81145628    Movie  Norm of the North: King Sized Adventure   
1  80117401    Movie               Jandino: Whatever it Takes   
2  70234439  TV Show                       Transformers Prime   
3  80058654  TV Show         Transformers: Robots in Disguise   
4  80125979    Movie                             #realityhigh   

                   director  \
0  Richard Finn, Tim Maltby   
1                       NaN   
2                       NaN   
3                       NaN   
4          Fernando Lebrija   

                                                cast  \
0  Alan Marriott, Andrew Toth, Brian Dobson, Cole...   
1                                   Jandino Asporaat   
2  Peter Cullen, Sumalee Montano, Frank Welker, J...   
3  Will Friedle, Darren Criss, Constance Zimmer, ...   
4  Nesta Cooper, Kate Walsh, John Michael Higgins...   

                                    country         date_added  release_year  \
0  United States, India, South Korea, China  September 9, 2019          2019   
1                            United Kingdom  September 9, 2016          2016   
2                             United States  September 8, 2018          2013   
3                             United States  September 8, 2018          2016   
4                             United States  September 8, 2017          2017   

     rating  duration                           listed_in  \
0     TV-PG    90 min  Children & Family Movies, Comedies   
1     TV-MA    94 min                     Stand-Up Comedy   
2  TV-Y7-FV  1 Season                            Kids' TV   
3     TV-Y7  1 Season                            Kids' TV   
4     TV-14    99 min                            Comedies   

                                         description  
0  Before planning an awesome wedding for his gra...  
1  Jandino Asporaat riffs on the challenges of ra...  
2  With the help of three human allies, the Autob...  
3  When a prison ship crash unleashes hundreds of...  
4  When nerdy high schooler Dani finally attracts...

In [5]:
movies.describe(include='all').head(4)

show_id   type        title                director  \
count    6234.0   6234         6234                    4265   
unique      NaN      2         6172                    3301   
top         NaN  Movie  The Silence  Raúl Campos, Jan Suter   
freq        NaN   4265            3                      18   

                      cast        country       date_added  release_year  \
count                 5664           5758             6223        6234.0   
unique                5469            554             1524           NaN   
top     David Attenborough  United States  January 1, 2020           NaN   
freq                    18           2032              122           NaN   

       rating  duration      listed_in  \
count    6224      6234           6234   
unique     14       201            461   
top     TV-MA  1 Season  Documentaries   
freq     2027      1321            299   

                                              description  
count                                                6234  
unique                                               6226  
top     A surly septuagenarian gets another chance at ...  
freq                                                    3

In [6]:
# Some cleaning
movies = movies.fillna('NULL')
movies['year_added'] = movies['date_added'].apply(lambda x: x.split(',')[-1])
movies['year_added'] = movies['year_added'].apply(lambda x : x if x != 'NULL' else '2020')
movies['year_added'] = movies['year_added'].apply(int)